### PR TITLE
Fix Colmap io utils

### LINKED ISSUE #527: Radial distortion ignored when loading colmap data
The implementation of `read_cameras_txt` in `io.py` ignores the radial distortion parameters k1 and k2 when parsing the lines of the `cameras.txt` file ([link](https://github.com/borglab/gtsfm/blob/f96bc07e99ad8b36e6f53c20be9221da97156eec/gtsfm/utils/io.py#L243))
Also, k1 and k2 have their values hardcoded to zero ([link](https://github.com/borglab/gtsfm/blob/f96bc07e99ad8b36e6f53c20be9221da97156eec/gtsfm/utils/io.py#L247))

### LINKED ISSUE #528: Sparse output in COLMAP format can't be used with COLMAP's GUI
The output generated by the `export_model_as_colmap_text` cannot be loaded in COLMAP's GUI. It crashes the application and generates a core dump.

It was identified that the `cameras.txt` file was not complying with the [colmap format](https://colmap.github.io/format.html) because the incompatibility between the number of parameters and camera model. 

According to the [colmap implementation](https://github.com/colmap/colmap/blob/d6f528ab59fd653966e857f8d0c2203212563631/src/base/camera_models.h) the following camera models are possible:
* RADIAL with 5 intrinsic parameters: f, cx, cy, k1, k2 
* SIMPLE_RADIAL with 4 intrinsic parameters:  f, cx, cy, k

[Current gtsfm implementation](https://github.com/borglab/gtsfm/blob/81f81af56cd38104de56a299357c3639a11b3c43/gtsfm/utils/io.py#L291) uses SIMPLE_RADIAL model with 5 parameters, which is incorrect.

### PR DESCRIPTION
**Pull request by luisfelipewb**
Some problems where observed with COLMAP io functions: 
1. Loss of radial distortion parameters.
2. Incompatible camera model tag on `cameras.txt` file.

This pull request includes:
* New unit tests for storing and loading camera data.
* Fix #527 
* Fix #528 
* Minor spelling and code style adjustments

IMPORTANT: only two camera models are supported after this change.

### CODE REVIEW

**Comment by luisfelipewb** [comment]
Adding link to issues:
fix #527
fix #528

**Comment by dellaert** [comment]
@gchenfc take a look at this PR for an example of a GTDFMData roundtrip, and possibly speed up this code in C++ by 100 :-)

File: tests/utils/test_io_utils.py [hunk 1]
```diff
@@ -84,6 +84,8 @@ def test_read_cameras_txt(self) -> None:
         self.assertEqual(K.fx(), 2332.47)
         self.assertEqual(K.px(), 2028)
         self.assertEqual(K.py(), 1520)
+        self.assertEqual(K.k1(), 0.00400066)
```
Comment (johnwlambert) [inline tests/utils/test_io_utils.py:87]: Could you add a small inline that we read in a COLMAP `SIMPLE_RADIAL` model (with 1 radial distortion coefficient), and implicitly converted it to GTSAM's "Cal3Bundler" (with 2 radial distortion coefficients). (The file doesn't contain a k2).

File: gtsfm/utils/io.py [hunk 2]
```diff
@@ -240,13 +237,26 @@ def read_cameras_txt(fpath: str) -> Optional[List[Cal3Bundler]]:
...
-        k1 = 0
-        k2 = 0
-        calibrations.append(Cal3Bundler(fx, k1, k2, u0, v0))
+        model = cam_params[1]
+        # Currently only handles SIMPLE RADIAL and RADIAL camera models
+        assert model in ["SIMPLE_RADIAL", "RADIAL"]
+        if model == "SIMPLE_RADIAL":
+            _, _, img_w, img_h, fx, u0, v0, k1 = cam_params[:8]
+            img_w, img_h, fx, u0, v0, k1 = int(img_w), int(img_h), float(fx), float(u0), float(v0), float(k1)
+            k2 = 0
```
Comment (johnwlambert) [inline gtsfm/utils/io.py:246]: Hi Luis, could you also add an inline comment here, saying that if the file contains a COLMAP `SIMPLE_RADIAL` model, we explicitly convert it to GTSAM's "Cal3Bundler" (adding a second radial distortion coefficient of value zero).

**Review by johnwlambert — APPROVED** [review]
Looks great, thanks for adding this. This situation occurred because we only ever imported COLMAP output, and only exported GTSFM output.

**Comment by luisfelipewb** [comment]
Thank you @johnwlambert for the review. I added the two comments as suggested in a new commit. It seems that the changes triggered the CI workflow that are awaiting approval.

**Pull request merged**